exercises/concept/weather-forecast-server/.docs/instructions.md

```diff
@@ -0,0 +1,10 @@
+# HTTP server
+
+Goblinocus president was happy with your previous work and now he demands your return to complete the weather forecast project! You are required to write an HTTP server that can respond to requests regarding the weather forecast for Goblinocus and an HTTP client that residents can use it.
+
+## Server
+
+ForecastServer must return a server that can appropriately handle clients' requests. You need to have a multiplexer that routes each request to its appropriate handler:
+
+1. if address ends with `“/city”`, it will send the request to a handler that sends back the forecast of the requested weather. Your handler must support only `http.Get` method. For any other method respond with MethodNotAllowed status code.
+2. For any other path, you need to use a handler to send back a BadRequest status code.
```

Comment:
As I have mentioned above, I was worried that maybe pushing more from the basics will be too much to ask for an exercise in this size. If you think it is OK I will add this to the exercise as well.

Reply:
Since you already split up the exercise in client and server, we have "space" now to cover some ground. Having 4-5 tasks in a concept exercise where each covers a different aspect is not uncommon.

If you want to start with an easy route, the first task could be about `/status` which simply returns "OK" or something like that. Then we should have a route with a city name as path parameter so that the student can practice extracting a path paramter. Don't worry about not having different responses. The response can be static or we can always make up something like "in the prototype version, the server returns a random number as temperature for the city" etc.

Also I really think GET and POST are ok to be in scope.

In any case, we need to make sure that the things people need to know for the exercise is covered in the introduction.

Reply:
I get your point. I'll add the discussed topics to the concept and the exercise.